internal/pkg/manifest/worker_svc.go

```diff
@@ -195,6 +202,29 @@ func (s *WorkerService) Subscriptions() []TopicSubscription {
 	return s.Subscribe.Topics
 }
 
+// RetrofitFIFOConfig appends ".fifo" suffix to fifo topics, also adds values to type field.
+func (s *WorkerService) RetrofitFIFOConfig() {
```

Comment:
Could we do all of this retrofiting work inside `UnmarshalYAML` so that we don't have to call separate `RetrofitFIFOConfig` method

```go
func (sub *TopicSubscription) UnmarshalYAML() error {
  // Set automatically the .fifo suffix to the topic name.
}

func (q *SQSQueue) UnmarshalYAML() error {
  // Set the queue type based on the user values.
}
```

Reply:
For each subscription, to determine whether it's `fifo` we need two pieces of information:
1. The type of the subscription's dedicated `queue`
2. The type of the default `queue`

We don't need 2 if we have 1.

So `UnmarshalYAML` might not work here 🤔  Because inside of 
```
func (sub *TopicSubscription) UnmarshalYAML() error { // Set automatically the .fifo suffix to the topic name. }
```
we can't know 2.

I'm thinking an alternative option is that we can have 
```
func (cfg *SubscribeConfig) Topics() []*TopicSubscription {
    // range over `cfg.Topics`, appending `.fifo` to `cfg.Topics[idx].Name` given `cfg.Topics[idx].Queue` or `cfg.Queue`
}

```

and then call `Topics` in `stack/transfomer.go`

Reply:
The only reason why we can not call this `Topics` in `stack/transfomer.go` is because there is a [validateTopicsExist](https://github.com/aws/copilot-cli/blob/4b82a0943e24c1db4c4f90e6a74b41928cb334e3/internal/pkg/cli/deploy/svc.go#L1177) function that checks for valid topics and if we do not retrofit before that point then that call fails for fifo topics.

Reply:
Then can we call it in `validateTopicsExist` 🤔  ` validateTopicsExist(d.wsMft.Subscriptions(), topicARNs, d.app.Name, d.env.Name`. We could maybe just modify the implementation of `Subscriptions`?